Branch-SDK/Branch-SDK/BNCServerInterface.m

```diff
@@ -138,10 +138,9 @@ - (void)genericHTTPRequest:(NSURLRequest *)request retryNumber:(NSInteger)retryN
             }
             
         }
-        dispatch_async(dispatch_get_main_queue(), ^{
-            if (callback)
-                callback(serverResponse, error);
-        });
+		//	Don't call on the main queue since it might be blocked.
```

Comment:
This callback block has traditionally always been handled on the main thread. Changing this could break a lot of existing integrations that might assume we're always calling their code on the main thread.